### PR TITLE
Fabric: Enable "auto" for `devices` and `accelerator` as cli arguments

### LINKED ISSUE #20451: Error: Invalid value for '--accelerator': 'auto' is not one of 'cpu', 'gpu', 'cuda', 'mps', 'tpu'.
### Bug description

when run this codes
```
fabric run ./path/to/train.py \
    --devices=auto \
    --accelerator=auto
```
get problem below
```
Error: Invalid value for '--accelerator': 'auto' is not one of 'cpu', 'gpu', 'cuda', 'mps', 'tpu'.
```

### What version are you seeing the problem on?

v2.4

### How to reproduce the bug

_No response_

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- PyTorch Lightning Version (e.g., 2.4.0):
#- PyTorch Version (e.g., 2.4):
#- Python version (e.g., 3.12):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
```

</details>

### More info

_No response_

### PR DESCRIPTION
**Pull request by fnhirwa**
## What does this PR do?

Fixes #20451 

This PR adds support for the `auto` variable for `accelerator` and `devices`  in the Fabric CLI. 

-  When `auto` is passed to `--accelerator` we now detect whether the machine has either `mps`, `cuda`, or `cpu` and dynamically runs on the available device, this is the same case when no accelerator is specified. 

- For `auto` in the `--devices` argument the devices default to `1` as specified in the docstring of the argument here https://github.com/Lightning-AI/pytorch-lightning/blob/c03660a8f84c7c1a09f3da230061b88e84430111/src/lightning/fabric/cli.py#L80

No breaking changes were introduced.

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20913.org.readthedocs.build/en/20913/

### CODE REVIEW

File: src/lightning/fabric/cli.py [hunk 1]
```diff
@@ -187,6 +187,19 @@ def _set_env_variables(args: Namespace) -> None:
 
 def _get_num_processes(accelerator: str, devices: str) -> int:
     """Parse the `devices` argument to determine how many processes need to be launched on the current machine."""
+    if accelerator == "auto" or accelerator is None:
```
Comment (Borda) [inline src/lightning/fabric/cli.py]: lets not duplicate the logic and use already written functions like:
https://github.com/Lightning-AI/pytorch-lightning/blob/c85660c0da7777c667309a3ef937276353aacc27/src/lightning/pytorch/trainer/connectors/accelerator_connector.py#L333

File: src/lightning/fabric/cli.py [hunk 2]
```diff
@@ -34,7 +34,18 @@
 _CLICK_AVAILABLE = RequirementCache("click")
 _LIGHTNING_SDK_AVAILABLE = RequirementCache("lightning_sdk")
 
-_SUPPORTED_ACCELERATORS = ("cpu", "gpu", "cuda", "mps", "tpu")
+_SUPPORTED_ACCELERATORS = ("cpu", "gpu", "cuda", "mps", "tpu", "auto")
+
+
+def _choose_auto_accelerator() -> str:
```
Comment (Borda) [inline src/lightning/fabric/cli.py]: so why do we have now the same function?

Comment (fnhirwa) [reply src/lightning/fabric/cli.py]: I tried to import from https://github.com/Lightning-AI/pytorch-lightning/blob/c85660c0da7777c667309a3ef937276353aacc27/src/lightning/pytorch/trainer/connectors/accelerator_connector.py#L333
But there were some dependencies errors, one for `torchmetrics` on some runners.

Comment (fnhirwa) [reply src/lightning/fabric/cli.py]: https://github.com/Lightning-AI/pytorch-lightning/actions/runs/15755765823/job/44410806075

Comment (Borda) [reply src/lightning/fabric/cli.py]: I see so them move the function to fabric and use it later in Connector
https://github.com/Lightning-AI/pytorch-lightning/tree/master/src/lightning/fabric/utilities

File: src/lightning/fabric/utilities/device_parser.py [hunk 3]
```diff
@@ -204,3 +204,18 @@ def _check_data_type(device_ids: object) -> None:
                 raise TypeError(f"{msg} a sequence of {type(id_).__name__}.")
     elif type(device_ids) not in (int, str):
         raise TypeError(f"{msg} {device_ids!r}.")
+
+
+def _select_auto_accelerator_fabric() -> str:
```
Comment (Borda) [inline src/lightning/fabric/utilities/device_parser.py]:
```suggestion
def _select_auto_accelerator() -> str:
```
